orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/pipeline/PipelineExpressionFunctionProvider.java

```diff
@@ -0,0 +1,97 @@
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.front50.pipeline;
+
+import com.netflix.spinnaker.orca.front50.Front50Service;
+import com.netflix.spinnaker.orca.pipeline.expressions.ExpressionFunctionProvider;
+import com.netflix.spinnaker.orca.pipeline.expressions.SpelHelperFunctionException;
+import com.netflix.spinnaker.orca.pipeline.model.Execution;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Map;
+
+@Component
+public class PipelineExpressionFunctionProvider implements ExpressionFunctionProvider {
+  /** Since expression evaluation happens in a static function we can't get access to front50 there
+   * Since this is a component it will be a singleton and hence safe to save the front50
+   * reference here
+   */
+  private static Front50Service front50Service;
+
+  @Autowired
+  PipelineExpressionFunctionProvider(Front50Service front50Service) {
+    PipelineExpressionFunctionProvider.front50Service = front50Service;
+  }
+
+
+  @Nullable
+  @Override
+  public String getNamespace() {
+    return null;
+  }
+
+  @NotNull
+  @Override
+  public Collection<FunctionDefinition> getFunctions() {
+    return Arrays.asList(
+      new FunctionDefinition("pipeline", Arrays.asList(
+        new FunctionParameter(
+          Execution.class, "execution", "The execution containing the currently executing stage"
+        ),
+        new FunctionParameter(
+          String.class, "pipelineName", "A valid stage reference identifier"
+        )
+      ))
+    );
+  }
+
+  /**
+   * Function to convert pipeline name to pipeline ID (within current application)
+   *
+   * @param execution the current execution
+   * @param pipelineName name of the pipeline to lookup
+   * @return the id of the pipeline or null if pipeline not found
+   */
+  public static String pipeline(Execution execution, String pipelineName) {
+    if ((pipelineName == null) || (pipelineName.isEmpty())) {
+      throw new SpelHelperFunctionException("pipelineName must be specified for function #pipeline");
+    }
+
+    if (front50Service == null) {
+      throw new SpelHelperFunctionException("front50 service is missing. It's required when using #pipeline function");
+    }
+
+    String currentApplication = execution.getApplication();
+
+    Map<String, Object> pipeline = front50Service.getPipelines(currentApplication)
```

Comment:
It might help to think about retries here (or alternative ways to be less susceptible to a network failure blowing up the pipeline execution).

Reply:
good point! is there a cache inside orca of pipelines? 'cuz we also eval expressions ALOT and if you have this function we will make the front50 call a bunch of times. maybe not really a problem just thinking out loud

Reply:
At a glance, I don't think there's a cache of pipelines in Front50. 

It _might_ not hurt to have a local background refreshed cache of application -> list of pipelines (name + id).

Reply:
I will a simple retry for now, we can talk about the cache for another PR